lib/auth/api.go

```diff
@@ -156,6 +160,7 @@ func NewWrapper(writer AccessPoint, cache ReadAccessPoint) AccessPoint {
 // and read operations are going though cache
 type Wrapper struct {
 	ReadAccessPoint
+	services.Semaphores
 	Write AccessPoint
 }
```

Comment:
So splitting the read and write halves here doesn't work like it does for the the other members of `AccessPoint` because we don't cache semaphores.  This is deliberate because semaphores are frequently modified, and all normal (non-`tctl`) semaphore operations are modifying operations, meaning that caching the semaphores would be a net loss in terms of performance.

The only reason `Wrapper` doesn't embed `AccessPoint` is because `AccessPoint` embeds `ReadAccessPoint`.  `Wrapper` is intercepting the read-only subset of methods on `AccessPoint` to divert them to the cache, and manually re-exporting the individual uncached methods.

Way I see it there are a few options here:
1. Don't embed `Semaphores`, and instead manually re-export all its methods (possibly renaming the `Write` field to `NoCache` in order to help clarify the real distinction here, since `Write.GetSemaphores(...)` is confusing).
2. Create a third `NoCacheAccessPoint` interface that is embedded in both `Wrapper` and `AccessPoint` just like `ReadAccessPoint` is, continuing to only manually export the methods which come from partially cached services (this is probably the most correct way to do things, but feels a bit silly since `Semaphores` would currently be the only member of `NoCacheAccessPoint`).
3. Keep it as is and add comment explaining that uncached services don't need to be subselected the way partially cached services do.  This is functionally the same as option 2, but possibly less convenient if we end up adding more uncached services down the road. 

Thoughts?

Reply:
no. 1 sounds great, it makes the intention here clearer.
maybe coupled with renaming `ReadAccesPoint` to `CacheableAccessPoint` or similar.